src/box/box.cc

```diff
@@ -831,8 +831,8 @@ wal_stream_apply_dml_row(struct wal_stream *stream, struct xrow_header *row)
 	assert(txn != NULL);
 	if (!row->is_commit)
 		return 0;
-	if (row->wait_ack)
-		box_txn_make_sync();
+	txn_set_xrow_flags(row->flags);
+	txn_recovery_set_limbo_policy();
```

Comment:
Lets just inline this function right here. It is used in a single place, and mentions "recovery", but doesn't depend on being called during the recovery anyhow. It also makes a check if called outside of any txns, which is never true here.